Main.py

```diff
@@ -2,431 +2,368 @@
-            msg += "[%s] %s: %d:%s (of %d)\n" % (scoreboard_len+1,rankings[i][0],minutes,str(seconds).zfill(2),score_count)
-        else:
-            msg += "[%s] %s: %d (of %d)\n" % (scoreboard_len+1,rankings[i][0],int(rankings[i][1]),score_count)
-        scoreboard_len += 1
-    msg += "```"
-    await self.bot.say(msg)
-    conn.close()
-
-@bot.command(pass_context=True)
-@commands.has_role("crosswords")
-@asyncio.coroutine
-async def saturdayrank(self):
+    await _rank(ctx, saturday=False)
+
+
+@bot.command()
+@commands.has_role(CROSSWORD_ROLE)
+async def saturdayrank(ctx):
     """
     Display the top 10 in the Saturday minicrossword scoreboard
     """
-    conn = sqlite3.connect(DB_PATH)
-    c = conn.cursor()
-    times = c.execute("SELECT Name,SatAvg FROM Ranking").fetchall()
-    if not times:
-        await self.bot.say("```No one has any Saturday crossword scores yet.```")
-        conn.close()
-        return
-    # if avg is None, set a default value so the sorted fn will still work
-    # need to create a new list since tuples are immutable
-    avgs = []
-    for i in range(len(times)):
-        if times[i][1] == None:
-            avgs.append((times[i][0],999999))
-        else:
-            avgs.append((times[i][0],times[i][1]))
-    rankings = sorted(avgs, key=lambda x:x[1])
-    msg = "```css\n"
-    msg += "Saturday Minicrossword Scoreboard:\n"
-    msg += "```"
-    await self.bot.say(msg)
-    msg = "```"
-
-    scoreboard_len = 0
-    for i in range(len(rankings)):
-        if rankings[i][1] == 999999:
-            continue
-
-        if scoreboard_len == 10:
-            break
-
-        unsorted_usr_timeslist = c.execute('SELECT Score,Date FROM Scores WHERE Name=?', (rankings[i][0],)).fetchall()
-        usr_timeslist = sorted(unsorted_usr_timeslist, key=lambda x:x[1])
-        score_count = 0
-        most_recent_day = ''
-        for t in usr_timeslist:
-            if datetime.datetime.strptime(t[1], '%Y-%m-%d').strftime('%a') == "Sat":
-                score_count += 1
-                most_recent_day = t[1]
-
-        delta = datetime.datetime.now() - datetime.datetime.strptime(most_recent_day, '%Y-%m-%d')
-        if delta.days > 30:
-            continue
-
-        if int(rankings[i][1]) > 59:
-            minutes = math.floor(int(rankings[i][1])/60)
-            seconds = int(rankings[i][1]) % 60
-            msg += "[%s] %s: %d:%s (of %d)\n" % (scoreboard_len+1,rankings[i][0],minutes,str(seconds).zfill(2),score_count)
-        else:
-            msg += "[%s] %s: %d (of %d)\n" % (scoreboard_len+1,rankings[i][0],int(rankings[i][1]),score_count)
-        scoreboard_len += 1
-    msg += "```"
-    await self.bot.say(msg)
-    conn.close()
-
-@bot.command(pass_context=True)
-@commands.has_role("crosswords")
-@asyncio.coroutine
-async def deltime(self):
+    await _rank(ctx, saturday=True)
+
+
+@bot.command()
+@commands.has_role(CROSSWORD_ROLE)
+async def deltime(ctx):
```

Comment:
average isn't updated when a score is deleted

Reply:
19e11bd1e01522040608edc461f68e8d7edf8cc6

Reply:
The average still isn't updated when a score is deleted

Reply:
Resolved in a0ffe74